go/service/assets/api_op_UpdateAssetDataAddress_test.go

```diff
@@ -0,0 +1,54 @@
+package assets
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/Think-iT-Labs/edc-connector-client/go/edc"
+	edchttp "github.com/Think-iT-Labs/edc-connector-client/go/edc/transport/http"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_UpdateAssetDataAddress(t *testing.T) {
+	authToken := "dummy"
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		payload, err := io.ReadAll(r.Body)
+		assert.NoError(t, err, "error while reading request body")
+		assert.JSONEq(t, `
+{
+	"edc:type":    "HttpData",
+	"edc:name":    "dummy asset server",
+	"edc:baseUrl": "http://dummy.dum"
+}`, string(payload), "invalid payload")
+		assert.Equal(t, "PUT", r.Method, "unexpected method")
+		assert.Equal(t, "/assets/1234/dataaddress", r.URL.Path)
+		w.WriteHeader(204)
```

Comment:
just to understand what's happening here, why are we returning a success status response code here if we know that the request isn't going through?

Reply:
This is an assert statement. It checks that the received payload is as expected. If it is not, testserver raises an "invalid payload" error. Otherwise, the next statement is executed (another assert).

Once everything is as expected, and no errors were raised, the testserver sends back a `204`.